chasm/tree.go

```diff
@@ -1540,6 +1555,56 @@ func (n *Node) closeTransactionGeneratePhysicalPureTask() error {
 	return nil
 }
 
+// resolveDeferredPointers resolves all deferred pointers in the tree.
+// Returns error if any deferred pointer cannot be resolved, as deferred pointers
+// cannot be persisted after transaction close.
+func (n *Node) resolveDeferredPointers() error {
+	for _, node := range n.andAllChildren() {
+		if node.value == nil || node.fieldType() != fieldTypeComponent {
+			continue
+		}
+
+		for field := range node.valueFields() {
+			if field.err != nil {
+				return field.err
+			}
+
+			if field.kind != fieldKindSubField {
+				continue
+			}
+
+			internalV := field.val.FieldByName(internalFieldName)
+			internal, _ := internalV.Interface().(fieldInternal) //nolint:revive
+
+			if internal.fieldType() == fieldTypeDeferredPointer && internal.value() != nil {
+				// Must resolve the deferred pointer or fail the transaction.
+				var resolvedPath []string
+				var err error
+
+				switch value := internal.value().(type) {
+				case Component:
+					resolvedPath, err = n.componentNodePath(value)
+				case proto.Message:
+					resolvedPath, err = n.dataNodePath(value)
```

Comment:
a bit expensive but should be fine I guess, we can optimize later by tracking all deserialized values if necessary.